flytekit/clis/auth/auth.py

```diff
@@ -305,3 +307,53 @@ def expired(self):
         :return bool:
         """
         return self._expired
+
+
+class GcpAuthorizationClient(object):
+    def __init__(self):
+        """
+         the gcp credentials will be fetched under two conditions:
+         1. If the environment variable ``GOOGLE_APPLICATION_CREDENTIALS`` is set
+        to the path of a valid service account JSON private key file
+         2. If the `Google Cloud SDK`_ is installed and has application default
+        credentials set they are loaded and returned.
+        In case of not able to get the credentials, the google lib will raise an exception
+        """
+        self._gcp_credentials, _ = _google_auth.default()
+        self._credentials = None
+        if isinstance(self._gcp_credentials, _service_account.Credentials):
+            target_audience = "https://flyte.net"
+            json_path = _os.environ.get("GOOGLE_APPLICATION_CREDENTIALS")
+            self._gcp_credentials = _service_account.IDTokenCredentials.from_service_account_file(
+                json_path, target_audience=target_audience
+            )
+
+    @property
+    def has_valid_credentials(self) -> bool:
+        return self._gcp_credentials is not None
+
+    @property
+    def can_id_token(self) -> bool:
+        return self._gcp_credentials is not None
+
+    def refresh_id_token(self):
+        auth_req = _grequests.Request()
```

Comment:
@EngHabu and @wild-endeavor Here is where token refresh happens. Basically you send a token refresh request along with client credentials as here: https://github.com/googleapis/google-auth-library-python/blob/82293fe2caaf5258babb5df1cff0a5ddc9e44b38/google/oauth2/credentials.py#L228